scout/parse/variant/variant.py

```diff
@@ -57,89 +57,66 @@ def parse_variant(
     # Vep information
     vep_header = vep_header or []
 
-    parsed_variant = {}
     # Create the ID for the variant
     case_id = case["_id"]
     genmod_key = get_genmod_key(case)
     chrom_match = CHR_PATTERN.match(variant.CHROM)
     chrom = chrom_match.group(2)
 
-    # Builds a dictionary with the different ids that are used
-    alt = get_variant_alternative(variant, category)
-
-    parsed_variant["ids"] = parse_ids(
-        chrom=chrom,
-        pos=variant.POS,
-        ref=variant.REF,
-        alt=alt,
-        case_id=case_id,
-        variant_type=variant_type,
-    )
-    parsed_variant["case_id"] = case_id
-    # type can be 'clinical' or 'research'
-    parsed_variant["variant_type"] = variant_type
-
-    category = get_category(category, variant, parsed_variant)
-    parsed_variant["category"] = category
-
-    ################# General information #################
-    parsed_variant["reference"] = variant.REF
-
-    ### We allways assume splitted and normalized vcfs!!!
+    ### We always assume split and normalized vcfs!!!
     if len(variant.ALT) > 1:
         raise VcfError("Variants are only allowed to have one alternative")
-    parsed_variant["alternative"] = alt
 
-    # cyvcf2 will set QUAL to None if '.' in vcf
-    parsed_variant["quality"] = variant.QUAL
+    # Builds a dictionary with the different ids that are used
+    alt = get_variant_alternative(variant, category)
+
+    coordinates = parse_coordinates(variant, category, case.get("genome_build"))
 
-    parsed_variant["filters"] = get_filters(variant)
+    parsed_variant = {
+        "ids": parse_ids(
+            chrom=chrom,
+            pos=variant.POS,
+            ref=variant.REF,
+            alt=alt,
+            case_id=case_id,
+            variant_type=variant_type,
+        ),
+        "case_id": case_id,
+        "variant_type": variant_type,
+        "reference": variant.REF,
+        "quality": variant.QUAL,  # cyvcf2 will set QUAL to None if '.' in vcf
+        "filters": get_filters(variant),
+        "alternative": alt,
+        "chromosome": chrom,
+        "cytoband_end": coordinates["cytoband_end"],
+        "cytoband_start": coordinates["cytoband_start"],
+        "end": coordinates["end"],
+        "end_chrom": coordinates["end_chrom"],
+        "length": coordinates["length"],
+        "mate_id": coordinates["mate_id"],
+        "position": coordinates["position"],
+        "sub_category": coordinates["sub_category"],
+        "samples": get_samples(variant, individual_positions, case, category),
```

Comment:
This is the main change really: we pass on the category given to this function for checking that the samples are relevant to the variant type.